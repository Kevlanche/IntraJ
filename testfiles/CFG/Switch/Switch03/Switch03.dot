digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[18]"; "Opt[19]"; "Exit[20]"}
{ rank = same; "List[15]"; "Block[20]"}
{ rank = same; "List[21]"}
{ rank = same; "SwitchStmt[22]"}
{ rank = same; "StringLiteral[23]"; "Block[24]"}
{ rank = same; "List[25]"}
{ rank = same; "ConstCase[26]"; "ExprStmt[28]"; "ConstCase[39]"; "ExprStmt[41]"; "BreakStmt[52]"}
{ rank = same; "StringLiteral[27]"; "Dot[29]"; "StringLiteral[40]"; "Dot[42]"; "List[54]"; "UncheckedExceptions[53]"}
{ rank = same; "Dot[30]"; "MethodAccess[33]"; "Dot[43]"; "MethodAccess[46]"}
{ rank = same; "TypeAccess[31]"; "VarAccess[32]"; "List[35]"; "List[37]"; "List[38]"; "UncheckedExceptions[34]"; "TypeAccess[44]"; "VarAccess[45]"; "List[48]"; "List[50]"; "List[51]"; "UncheckedExceptions[47]"}
{ rank = same; "StringLiteral[36]"; "StringLiteral[49]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < Switch03 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[19]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[20]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[21]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SwitchStmt[22]" [label="SwitchStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[23]" [label="StringLiteral\n < bar >", fillcolor=white   style=filled  ]
"Block[24]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[25]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstCase[26]" [label="ConstCase\n < foo >", fillcolor=white   style=filled  ]
"StringLiteral[27]" [label="StringLiteral\n < foo >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[28]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[29]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[30]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[31]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[32]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[33]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[36]" [label="StringLiteral\n < foo >", fillcolor=white   style=filled  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[34]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstCase[39]" [label="ConstCase\n < bar >", fillcolor=white   style=filled  ]
"StringLiteral[40]" [label="StringLiteral\n < bar >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[41]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[42]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[43]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[44]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[45]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[46]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[48]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[49]" [label="StringLiteral\n < bar >", fillcolor=white   style=filled  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[51]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[47]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BreakStmt[52]" [label="BreakStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[54]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[53]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[20]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[18]" [style=dashed, color=gray]
"SwitchStmt[22]" -> "StringLiteral[23]" [style=dashed, color=gray]
"ConstCase[26]" -> "StringLiteral[27]" [style=dashed, color=gray]
"List[25]" -> "ConstCase[26]" [style=dashed, color=gray]
"Dot[30]" -> "TypeAccess[31]" [style=dashed, color=gray]
"Dot[30]" -> "VarAccess[32]" [style=dashed, color=gray]
"Dot[29]" -> "Dot[30]" [style=dashed, color=gray]
"List[35]" -> "StringLiteral[36]" [style=dashed, color=gray]
"MethodAccess[33]" -> "List[35]" [style=dashed, color=gray]
"MethodAccess[33]" -> "List[37]" [style=dashed, color=gray]
"MethodAccess[33]" -> "List[38]" [style=dashed, color=gray]
"MethodAccess[33]" -> "UncheckedExceptions[34]" [style=dashed, color=gray]
"Dot[29]" -> "MethodAccess[33]" [style=dashed, color=gray]
"ExprStmt[28]" -> "Dot[29]" [style=dashed, color=gray]
"List[25]" -> "ExprStmt[28]" [style=dashed, color=gray]
"ConstCase[39]" -> "StringLiteral[40]" [style=dashed, color=gray]
"List[25]" -> "ConstCase[39]" [style=dashed, color=gray]
"Dot[43]" -> "TypeAccess[44]" [style=dashed, color=gray]
"Dot[43]" -> "VarAccess[45]" [style=dashed, color=gray]
"Dot[42]" -> "Dot[43]" [style=dashed, color=gray]
"List[48]" -> "StringLiteral[49]" [style=dashed, color=gray]
"MethodAccess[46]" -> "List[48]" [style=dashed, color=gray]
"MethodAccess[46]" -> "List[50]" [style=dashed, color=gray]
"MethodAccess[46]" -> "List[51]" [style=dashed, color=gray]
"MethodAccess[46]" -> "UncheckedExceptions[47]" [style=dashed, color=gray]
"Dot[42]" -> "MethodAccess[46]" [style=dashed, color=gray]
"ExprStmt[41]" -> "Dot[42]" [style=dashed, color=gray]
"List[25]" -> "ExprStmt[41]" [style=dashed, color=gray]
"BreakStmt[52]" -> "List[54]" [style=dashed, color=gray]
"BreakStmt[52]" -> "UncheckedExceptions[53]" [style=dashed, color=gray]
"List[25]" -> "BreakStmt[52]" [style=dashed, color=gray]
"Block[24]" -> "List[25]" [style=dashed, color=gray]
"SwitchStmt[22]" -> "Block[24]" [style=dashed, color=gray]
"List[21]" -> "SwitchStmt[22]" [style=dashed, color=gray]
"Block[20]" -> "List[21]" [style=dashed, color=gray]
"Opt[19]" -> "Block[20]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[19]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[20]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[20]" -> "ConstCase[39]" [color=red, constraint=false]
"Exit[20]" -> "MethodAccess[46]" [color=red, constraint=false]
"MethodAccess[46]" -> "Exit[20]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[46]" -> "StringLiteral[49]" [color=red, constraint=false]
"StringLiteral[49]" -> "MethodAccess[46]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[49]" -> "VarAccess[45]" [color=red, constraint=false]
"VarAccess[45]" -> "StringLiteral[49]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[45]" -> "TypeAccess[44]" [color=red, constraint=false]
"TypeAccess[44]" -> "VarAccess[45]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[44]" -> "MethodAccess[33]" [color=red, constraint=false]
"TypeAccess[44]" -> "ConstCase[39]" [color=red, constraint=false]
"MethodAccess[33]" -> "TypeAccess[44]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[33]" -> "StringLiteral[36]" [color=red, constraint=false]
"StringLiteral[36]" -> "MethodAccess[33]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[36]" -> "VarAccess[32]" [color=red, constraint=false]
"VarAccess[32]" -> "StringLiteral[36]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[32]" -> "TypeAccess[31]" [color=red, constraint=false]
"TypeAccess[31]" -> "VarAccess[32]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[31]" -> "ConstCase[26]" [color=red, constraint=false]
"ConstCase[39]" -> "TypeAccess[44]" [color=blue, constraint=false, xlabel="T "] 
"ConstCase[39]" -> "Exit[20]" [color=blue, constraint=false, xlabel="F "] 
"ConstCase[39]" -> "ConstCase[26]" [color=red, constraint=false]
"ConstCase[26]" -> "TypeAccess[31]" [color=blue, constraint=false, xlabel="T "] 
"ConstCase[26]" -> "ConstCase[39]" [color=blue, constraint=false, xlabel="F "] 
"ConstCase[26]" -> "StringLiteral[23]" [color=red, constraint=false]
"StringLiteral[23]" -> "ConstCase[26]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[23]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "StringLiteral[23]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[18]" -> "Opt[19]" -> "Exit[20]" [style=invis]
"StringLiteral[23]" -> "Block[24]" [style=invis]
"ConstCase[26]" -> "ExprStmt[28]" -> "ConstCase[39]" -> "ExprStmt[41]" -> "BreakStmt[52]" [style=invis]
"Dot[30]" -> "MethodAccess[33]" [style=invis]
"TypeAccess[31]" -> "VarAccess[32]" [style=invis]
"Dot[43]" -> "MethodAccess[46]" [style=invis]
"TypeAccess[44]" -> "VarAccess[45]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
56", fillcolor=1]
Edges [label="Total Nr. Edges
83", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
13", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
14", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}