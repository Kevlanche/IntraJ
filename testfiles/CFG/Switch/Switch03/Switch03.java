public class Switch03 {
  void m() {
    switch ("bar") {
    case "foo":
      System.out.println("foo");
    case "bar":
      System.out.println("bar");
      break;
    }
  }
}