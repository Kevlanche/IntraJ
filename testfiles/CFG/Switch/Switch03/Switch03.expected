NodeName: Entry[13] Successors: [StringLiteral[23]]
NodeName: StringLiteral[23] Successors: [ConstCase[26]]
NodeName: ConstCase[26] Successors: [TypeAccess[31], ConstCase[39]]
NodeName: TypeAccess[31] Successors: [VarAccess[32]]
NodeName: VarAccess[32] Successors: [StringLiteral[36]]
NodeName: StringLiteral[36] Successors: [MethodAccess[33]]
NodeName: MethodAccess[33] Successors: [TypeAccess[44]]
NodeName: TypeAccess[44] Successors: [VarAccess[45]]
NodeName: VarAccess[45] Successors: [StringLiteral[49]]
NodeName: StringLiteral[49] Successors: [MethodAccess[46]]
NodeName: MethodAccess[46] Successors: [Exit[20]]
NodeName: Exit[20] Successors: []
NodeName: ConstCase[39] Successors: [TypeAccess[44], Exit[20]]

NodeName: Exit[20] Predecessor: [ConstCase[39], MethodAccess[46]]
NodeName: ConstCase[39] Predecessor: [ConstCase[26]]
NodeName: ConstCase[26] Predecessor: [StringLiteral[23]]
NodeName: StringLiteral[23] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: MethodAccess[46] Predecessor: [StringLiteral[49]]
NodeName: StringLiteral[49] Predecessor: [VarAccess[45]]
NodeName: VarAccess[45] Predecessor: [TypeAccess[44]]
NodeName: TypeAccess[44] Predecessor: [MethodAccess[33], ConstCase[39]]
NodeName: MethodAccess[33] Predecessor: [StringLiteral[36]]
NodeName: StringLiteral[36] Predecessor: [VarAccess[32]]
NodeName: VarAccess[32] Predecessor: [TypeAccess[31]]
NodeName: TypeAccess[31] Predecessor: [ConstCase[26]]
